opencti-platform/opencti-graphql/src/modules/catalog/catalog-domain.ts

```diff
@@ -19,22 +19,20 @@ addFormats(ajv, ['password', 'uri', 'duration', 'email', 'date-time', 'date']);
 let catalogMap: Record<string, CatalogType>;
 // cache for contracts by image map
 let contractsByImageCache: Map<string, CatalogContract> | undefined;
-// Bypass catalog cache flag - when enabled, catalogs are loaded without cache
-let bypassCatalogCache = false;
 
 // Build catalog map from files
 const buildCatalogMap = (): Record<string, CatalogType> => {
   const newCatalogMap: Record<string, CatalogType> = {};
-
-  const catalogs = CUSTOM_CATALOGS.map((custom) => fs.readFileSync(custom, { encoding: 'utf8', flag: 'r' }));
+  const catalogs = [];
   catalogs.push(JSON.stringify(filigranCatalog));
-
+  catalogs.push(...CUSTOM_CATALOGS.map((custom) => fs.readFileSync(custom, { encoding: 'utf8', flag: 'r' })));
```

Comment:
readFileSync blocks the  event loop, use 'fs/promise' method readFile instead

Reply:
Done